pkg/cmd/issue/shared/lookup.go

```diff
@@ -19,26 +19,61 @@ import (
 // IssueFromArgWithFields loads an issue or pull request with the specified fields. If some of the fields
 // could not be fetched by GraphQL, this returns a non-nil issue and a *PartialLoadError.
 func IssueFromArgWithFields(httpClient *http.Client, baseRepoFn func() (ghrepo.Interface, error), arg string, fields []string) (*api.Issue, ghrepo.Interface, error) {
-	issueNumber, baseRepo := issueMetadataFromURL(arg)
+	issues, baseRepo, err := IssuesFromArgsWithFields(httpClient, baseRepoFn, []string{arg}, fields)
+	var issue *api.Issue
+	if len(issues) == 1 {
+		issue = issues[0]
+	} else if len(issues) > 1 {
+		return nil, nil, fmt.Errorf("multiple issues not expected")
+	}
+	return issue, baseRepo, err
+}
 
-	if issueNumber == 0 {
-		var err error
-		issueNumber, err = strconv.Atoi(strings.TrimPrefix(arg, "#"))
-		if err != nil {
-			return nil, nil, fmt.Errorf("invalid issue format: %q", arg)
-		}
+// IssuesFromArgWithFields loads 1 oe more issues or pull requests with the specified fields. If some of the fields
+// could not be fetched by GraphQL, this returns non-nil issues and a *PartialLoadError.
+func IssuesFromArgsWithFields(httpClient *http.Client, baseRepoFn func() (ghrepo.Interface, error), args []string, fields []string) ([]*api.Issue, ghrepo.Interface, error) {
```

Comment:
This feels a bit backwards with `IssueFromArgWithFields` calling `IssuesFromArgWithFields`. It feels like it should be revered where `IssuesFromArgWithFields` is just a loop calling `IssueFromArgWithFields` with some concurrency built in.

Reply:
Yeah, hadn't considered that. Does seem more elegant.